students/khtruong/lesson_05/mailroom_v3.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python
+from datetime import date
+
+"""This module runs the mailroom script.
+"""
+
+data_dict = {'Jeff Bezos': [3.65, 54.50],
+             'Mark Zuckerberg': [36.54, 1.25, 54.87],
+             'Paul Allen': [17.38],
+             'William Gates III': [25.55, 33.33, 78.14]
+             }
+
+
+def menu_selection(prompt, selection_dict):
+    """Return options for user to select from."""
+    while True:
+        userinput = input(prompt)
+        if selection_dict.get(userinput) is None:
+            print('\n{} is not a valid selection. '
+                  'Please try again!'.format(userinput))
+        else:
+            if selection_dict.get(userinput)() == 'Exit Menu':
+                break
+
+
+def exit_menu():
+    """Return 'Exit Menu'."""
+    return 'Exit Menu'
+
+
+def send_thankyou_email():
+    """Return a menu selection to send thank you email to donor."""
+    menu_selection(thankyou_prompt, thankyou_dict)
+
+
+def list_donors():
+    """Return all donors name."""
+    [print(donor) for donor in data_dict.keys()]
+
+
+def donor_and_amount():
+    """Return prompt for donor name and donation amount and print to screen."""
+    fullname = fullname_input()
+    amount = amount_input()
+    data_dict.setdefault(fullname, [])  # add name to key if doesn't exist
+    data_dict[fullname].append(amount)
+    print(format_letter(fullname, amount))
+
+
+def amount_input():
+    """Return prompt asking for donation amount."""
+    while True:
+        try:
+            amount = float(input('Enter donation amount! > '))
+        except ValueError:
+            print('\nPlease enter dollar amount and NOT text!')
+        else:
+            break
+    return amount
+
+
+def fullname_input():
+    """Return prompt asking for full name."""
+    return input('Enter a donor first and last name > ').title()
+
+
+def summarize_donation():
+    '''Return total given, number of gifts, and average gifts for each donor'''
+    summarized_list = [{'donor_name': donor_key,
+                        'total_given': sum(data_dict.get(donor_key)),
+                        'num_gifts': len(data_dict.get(donor_key)),
+                        'avg_gifts': (sum(data_dict.get(donor_key)) /
+                                      max(len(data_dict.get(donor_key)), 1))
+                        } for donor_key in data_dict.keys()]
```

Comment:
nice job on utilizing list comprehension. here are some styling suggestions:

```
summarized_list = [{"donor_name": k, "total_given": sum(v), "num_gifts": len(v), "avg_gifts": sum(v) / len(v)} for k, v
                   in data_dict.items()]
```

use `items` if you need both key and value. it's ok to use shorter variables in comprehensions